### PR TITLE
Add `cookiejar` types and improve `superagent`

### PR DESCRIPTION
**Pull request by paroxp**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] ~~Create it with `dts-gen --dt`, not by basing it on an existing project.~~
  - I winged it myself. sorry :(
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/visionmedia/superagent/blob/master/lib/node/agent.js#L31
- [ ] ~~Increase the version number in the header if appropriate.~~
- [ ] ~~If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.~~

### CODE REVIEW

**Comment by paroxp** [comment]
Sorry about the `referenced this pull request` spam. It may be recurring upon rebasing :(

On the bright side, it's the proof of working code that is compiled and tested on travis!

File: types/cookiejar/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    getCookie(cookieName: string, accessInfo: CookieAccessInfo): Cookie;
+
+    /**
+     * grab all cookies matching this access_info
+     * @param accessInfo CookieAccessInfo
+     */
+    getCookies(accessInfo: CookieAccessInfo): ReadonlyArray<Cookie>;
+}
+
+export default CookieJar;
```
Comment (mhegazy) [inline types/cookiejar/index.d.ts]: At runtime this package does not have a default export. see:

```shell
c:\test\dt>npm install cookiejar
+ cookiejar@2.1.2
added 1 package from 1 contributor and audited 63 packages in 2.015s

c:\test\dt>node
> var c = require("cookiejar");
undefined
> c;
{ CookieAccessInfo: { [Function: CookieAccessInfo] All: {} },
  Cookie: [Function: Cookie],
  CookieJar: [Function: CookieJar] }
```
Please see https://github.com/DefinitelyTyped/DefinitelyTyped#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default for mode details.

**Review by mhegazy — CHANGES_REQUESTED** [review]
Please remove the `export default CookieJar`

**Comment by paroxp** [comment]
My bad, removed.

Thanks.

**Comment by mhegazy** [comment]
Thanks. There are some lint failures that needs attention.

**Comment by paroxp** [comment]
Yeah, done that too :) It's all green now.

Thanks!

**Pull request merged**